hapi-fees/src/main/java/com/hedera/services/calc/OverflowCheckingCalc.java

```diff
@@ -0,0 +1,124 @@
+package com.hedera.services.calc;
+
+/*-
+ * ‌
+ * Hedera Services API Fees
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.usage.state.UsageAccumulator;
+import com.hederahashgraph.api.proto.java.ExchangeRate;
+import com.hederahashgraph.api.proto.java.FeeComponents;
+import com.hederahashgraph.api.proto.java.FeeData;
+import com.hederahashgraph.fee.FeeBuilder;
+import com.hederahashgraph.fee.FeeObject;
+
+import static com.hedera.services.usage.SingletonEstimatorUtils.ESTIMATOR_UTILS;
+import static com.hederahashgraph.fee.FeeBuilder.FEE_DIVISOR_FACTOR;
+
+public class OverflowCheckingCalc {
+	private static final String POSITIVITY_ERROR = "All fee contributions and intermediate terms must be positive";
+
+	public FeeObject fees(UsageAccumulator usage, FeeData prices, ExchangeRate rate, long multiplier) {
+		final long networkFeeTinycents = networkFeeInTinycents(usage, prices.getNetworkdata());
+		final long nodeFeeTinycents = nodeFeeInTinycents(usage, prices.getNodedata());
+		final long serviceFeeTinycents = serviceFeeInTinycents(usage, prices.getServicedata());
+
+		final long networkFee = tinycentsToTinybars(networkFeeTinycents, rate) * multiplier;
+		final long nodeFee = tinycentsToTinybars(nodeFeeTinycents, rate) * multiplier;
+		final long serviceFee = tinycentsToTinybars(serviceFeeTinycents, rate) * multiplier;
+
+		if (networkFee < 0 || nodeFee < 0 || serviceFee < 0) {
+			throw new IllegalArgumentException(POSITIVITY_ERROR);
```

Comment:
Does `0` (zero) belong to positive numbers?

Reply:
Good point @ljianghedera, we should rename it to `NON_NEGATIVITY_ERROR` and change the string content.